File: CD/projeto/APIWeb/gis/src/main/java/pt/isel/ps/gis/hypermedia/siren/components/subentities/Entity.java
package pt.isel.ps.gis.hypermedia.siren.components.subentities;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.util.HashMap;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({"class", "rel", "properties", "href"})
public class Entity {
    @JsonProperty(value = "class")
    public String[] klass;
    public String[] rel;
    public HashMap<String, Object> properties;
    public Action[] actions;
    public Link[] links;

    public Entity(String[] klass, String[] rel, HashMap<String, Object> properties, Action[] actions, Link[] links) {
        this.klass = klass;
        this.rel = rel;
        this.properties = properties;
        this.actions = actions;
        this.links = links;
    }
}

File: Android/gis/app/src/main/java/ps/leic/isel/pt/gis/hypermedia/subentities/Entity.kt
package ps.leic.isel.pt.gis.hypermedia.subentities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.util.HashMap;

@JsonIgnoreProperties(ignoreUnknown = true)
class Entity(@field:JsonProperty(value = "class") var klass: Array<String>?, var rel: Array<String>,
             var properties: HashMap<String, Any>?, var actions: Array<Action>?, var links: Array<Link>)
